WordPressKit/TransactionsServiceRemote.swift

```diff
@@ -36,77 +36,58 @@ import WordPressShared
         })
     }
 
-    /// Creates a shopping cart for a domain purchase
+    /// Creates a shopping cart with products
     /// - Parameters:
     ///   - siteID: id of the current site
-    ///   - domainSuggestion: suggested new domain to purchase
+    ///   - products: an array of products to be added to the newly created cart
     ///   - temporary: true if the card is temporary, false otherwise
-    ///   - privacyProtectionEnabled: true if privacy protection on the given domain is enabled
-    private func createDomainShoppingCart(siteID: Int,
-                                          domainSuggestion: DomainSuggestion,
-                                          privacyProtectionEnabled: Bool,
-                                          temporary: Bool,
-                                          success: @escaping (CartResponse) -> Void,
-                                          failure: @escaping (Error) -> Void) {
+    public func createShoppingCart(siteID: Int,
+                                   products: [TransactionsServiceProduct],
+                                   temporary: Bool,
+                                   success: @escaping (CartResponse) -> Void,
+                                   failure: @escaping (Error) -> Void) {
 
         let endPoint = "me/shopping-cart/\(siteID)"
         let urlPath = path(forEndpoint: endPoint, withVersion: ._1_1)
 
-        var productDictionary: [String: AnyObject] = ["product_id": domainSuggestion.productID as AnyObject,
-                                                      "meta": domainSuggestion.domainName as AnyObject]
+        var productsDictionary: [[String: AnyObject]] = []
 
-        if privacyProtectionEnabled {
-            productDictionary["extra"] = ["privacy": true] as AnyObject
+        for product in products {
+            switch product {
+            case .domain(let domainSuggestion, let privacyProtectionEnabled):
+                productsDictionary.append(["product_id": domainSuggestion.productID as AnyObject,
+                                           "meta": domainSuggestion.domainName as AnyObject,
+                                           "extra": ["privacy": privacyProtectionEnabled] as AnyObject])
+
+            case .plan(let productId):
+                productsDictionary.append(["product_id": productId as AnyObject])
```

Comment:
@ravishanker, @irfano I found it enough to pass "product_id" to add a plan to a cart.